### PR TITLE
Add documentation for triple-slash-reference-types directives

### PR DESCRIPTION
**Pull request by mhegazy**
Fixes https://github.com/Microsoft/TypeScript-Handbook/issues/348

### CODE REVIEW

File: pages/Triple-Slash Directives.md [hunk 1]
```diff
@@ -37,6 +37,23 @@ It is an error for a file to have a triple-slash reference to itself.
...
+Similar to a `/// <reference path="..." />` directive,  this directive serves as a declaration of *dependency*;
+a `/// <reference types="..." />`, however, declares a dependency on an `@types` package.
+
+Including `/// <reference types="node" />` in a declaration file declares that this file uses names declared in `@types/node/index.d.ts`;
+and thus, this package needs to be included in the compilation along with the declaration file.
+
+The process of resolving these `@types` package names are similar to process of resolving module names in an `import` statement.
+An easy way to think of triple-slash-reference-types directives as an `import` for declaration packages.
+
+Also worth noting that the compiler will automatically add `/// <reference types="..." />` in generated declaration files if the resulting files uses any declarations from an `@types` package.
```
Comment (sandersn) [inline pages/Triple-Slash Directives.md]: I would add "It's" at the beginning

Comment (sandersn) [inline pages/Triple-Slash Directives.md]: typo: if the resulting files **use**

File: pages/Triple-Slash Directives.md [hunk 2]
```diff
@@ -37,6 +37,23 @@ It is an error for a file to have a triple-slash reference to itself.
...
+
+These directives should not be used unless you are authoring a `.d.ts` file by hand.
```
Comment (sandersn) [inline pages/Triple-Slash Directives.md]: I would move this just after the first sentence, and put positively: "Use these directives only when you're authoring a `d.ts` file by hand."

Then maybe another sentence explaining why they're needed here and nowhere else just after the *second* sentence (or so).

File: pages/Triple-Slash Directives.md (hunk 1 above)
Comment (mhegazy) [inline pages/Triple-Slash Directives.md]: both fixed.

**Comment by mhegazy** [comment]
@sandersn can you take another look? i have moved the .d.ts sentence around a bit.

File: pages/Triple-Slash Directives.md [hunk 3]
```diff
@@ -37,6 +37,25 @@ It is an error for a file to have a triple-slash reference to itself.
...
+
+Including `/// <reference types="node" />` in a declaration file declares that this file uses names declared in `@types/node/index.d.ts`;
+and thus, this package needs to be included in the compilation along with the declaration file.
+
+The process of resolving these `@types` package names are similar to process of resolving module names in an `import` statement.
+An easy way to think of triple-slash-reference-types directives as an `import` for declaration packages.
+
+Use these directives only when you're authoring a `d.ts` file by hand.
+
+For declaration files generatd during compilation, the compiler will automatically add `/// <reference types="..." />` for you;
```
Comment (sandersn) [inline pages/Triple-Slash Directives.md]: typo:generated

**Pull request merged**